### ISSUE TITLE
value of this._player lost in nativescript-audio player class

### ISSUE THREAD

**Issue by jokro** [nstudio/nativescript-audio#138]
I am using the audio-player example  with code below. When calling method play, it turns out that this._recorder is null, hence playing is not possible. I have inserted rules (see comment TEST) which show that this._player still has its value at the end of construtor. Why is the value of this._player lost at play time?
When I remove this._player outside the class (define var recorder and use it in stead) all works fine. This is no solution however, because I want to be able to have two players (each player must remember its state when paused).

The ts file:
```typescript

import { TNSPlayer } from 'nativescript-audio';
 
export class Player {
  private _player: TNSPlayer;
 
  constructor() {
    this._player = new TNSPlayer();

    this._player.debug = true; // set true to enable TNSPlayer console logs for debugging.
    this._player
      .initFromFile({
        audioFile: '~/audio/song.mp3', // ~ = app directory
        loop: false,
        completeCallback: this._trackComplete.bind(this),
        errorCallback: this._trackError.bind(this)
      })
      .then(() => {
        this._player.getAudioTrackDuration().then(duration => {
          // iOS: duration is in seconds
          // Android: duration is in milliseconds
          console.log(`song duration:`, duration);
        });
      });

      if(!this._player) alert ('Constructor END this._player undefined') //TEST
  }
 
  public play() {
      console.log('PLAY')
      if(!this._player) alert ('PLAY this._player undefined')  //TEST
    if (this._player)       this._player.play();
}

public pause() {
    if(!this._player) alert ('PAUSE this._player undefined')  //TEST
    if (this._player)       this._player.pause();
}

  public togglePlay() {
    if (this._player.isAudioPlaying()) {
      this._player.pause();
    } else {
      this._player.play();
    }
  }
 
  private _trackComplete(args: any) {
    console.log('reference back to player:', args.player);
    // iOS only: flag indicating if completed succesfully
    console.log('whether song play completed successfully:', args.flag);
  }
 
  private _trackError(args: any) {
    console.log('reference back to player:', args.player);
    console.log('the error:', args.error);
    // Android only: extra detail on error
    console.log('extra info on the error:', args.extra);
  }
}

const player1=new Player()
export const player_play=player1.play
export const player_pause=player1.pause
export const player_toggle=player1.togglePlay
```
xml file:

```
<Page xmlns="http://schemas.nativescript.org/tns.xsd" navigatingTo="onNavigatingTo">
<StackLayout>
<Button text="Start Play" tap="player_play" />
<Button text="Stop Play" tap="player_pause" />
</StackLayout>
</Page>
```

**Comment by bradmartin** [comment]
Looks like you forgot to include some code. How are you creating the instance of your player class?

Also in your player class you only have one tns_player declared as a member of the class. You mention having two players so you'd need to create two separate instances of TNS_Player. 

Maybe there is something I'm not understanding though 😁

**Comment by jokro** [comment]
I do not see your point that I did not create an instance of player. At the end of the ts file I created an instance player1 with

```
const player1=new Player()
export const player_play=player1.play
etc.
```

I did not create a second instance in this example because this is not necessary to show the problem at hand.  This problem arises in the code for doing the play:

```
public play() {
      console.log('PLAY')
      if(!this._player) alert ('PLAY this._player undefined')  //TEST
    if (this._player)       **_this._player.play();_**
}
```

In this, **_this._player.play()_** gives an error because this._player is null. By _**player1=new Player()**_ the constructor of the class defines  _**this._player = new TNSPlayer();**_. This defined value should still exist in method play described above. I hope I made my problem. Otherwise you could use the xml and ts code to test it yourself. If you press button play an alert will  show that this._player is null, while it should be an instance of TNSPlayer.

**Comment by bradmartin** [comment]
Are you certain the code is executed in your .ts file? Typically, in NS your "code-behind" file uses some event from the template/page. I see `onNavigatingTo` event on the `Page` in your template but that event is not in your .ts code it doesn't seem.

Have you tried creating an instance of your player during this `Page` event?

**Comment by bradmartin** [comment]
In the meantime, can you create a repo with this logic and I'd be happy to take a look.

Thanks.

**Comment by jokro** [comment]
Yes I am sure the code is executed. I am running it in the debugger and can follow the steps of initialising Player and executing start. 

In this example onNavigatingTo is not functional, therfore I did not include it. I did test it with the following code however, showing that the page is loaded before started.

```
export const onNavigatingTo=(args)=> {
    console.log(" onNavigatingTo LOADED")  
 }
```

If you want to test it, just copy te above code. Unfortunatily I cannot put it into playground as it does not recognize the import of nativescript-audio.
